src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/service/AsyncGoalServiceDecorator.kt

```diff
@@ -0,0 +1,34 @@
+package uk.gov.justice.digital.hmpps.educationandworkplanapi.app.service
+
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.runBlocking
+import org.springframework.stereotype.Component
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.domain.goal.Goal
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.domain.goal.dto.CreateGoalDto
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.domain.goal.dto.UpdateGoalDto
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.domain.goal.service.GoalServiceDecorator
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.domain.goal.service.NoopGoalServiceDecorator
+
+/**
+ * Implementation of [GoalServiceDecorator] that decorates Goal Service methods with additional behaviours that are
+ * invoked asynchronously.
+ */
+@Component
+class AsyncGoalServiceDecorator(private val telemetryService: TelemetryService) : NoopGoalServiceDecorator() {
```

Comment:
I've not written a unit test for this class yet - I can do if we decide its the right design approach

Reply:
You didn't write the test first?! ;-)

Minor point, but I'm not sure about extending the Noop version. I would just implement the original interface :)